src/wrapper.jl

```diff
@@ -110,9 +128,40 @@ macro lightgbm(f, params...)
     end
 end
 
+
 # function LGBM_DatasetCreateFromFile()
-# function LGBM_DatasetCreateFromCSR()
-# function LGBM_DatasetCreateFromCSC()
+
+
+function LGBM_DatasetCreateFromCSC(
+    data::SparseArrays.SparseMatrixCSC,
+    parameters::String,
+    reference::Dataset = Dataset(C_NULL),
+)
+
+    if data.m > typemax(Int32)
+        throw(DomainError(data.m, "Cannot accept CSC matrices with more than $(typemax(Int32)) rows"))
+    end
+
+    idx_type, data_type = sparsetypes(data)
+
+    out = Ref{DatasetHandle}()
+    @lightgbm(
+        :LGBM_DatasetCreateFromCSC,
+        data.colptr .- 1 => Ptr{Nothing},
+        idx_type => Cint,
+        Int32.(data.rowval .- 1) => Ptr{Cint},
+        data.nzval => Ptr{Nothing},
+        data_type => Cint,
```

Comment:
data_type – Type of data pointer, can be C_API_DTYPE_FLOAT32 or C_API_DTYPE_FLOAT64 shouldn't be this type?

Reply:
`data_type` is a variable, which is one of `C_API_DTYPE_FLOAT32` `C_API_DTYPE_FLOAT64` (determined by `sparsedatatype` call, see https://github.com/IQVIA-ML/LightGBM.jl/pull/113/files#diff-e08580f7ddadf1983cf458c25618ce250ee72b50d5924cd1375df3c4b023d963R102) 
`Cint` is the type that needs to be passed to the FFI inteface (under the hood, `ccall`) 

So what we're doing here is passing a C integer, which takes a value encoding F32 or F64, and telling `ccall` that this variable is a C integer